artiq/frontend/artiq_coremgmt.py

```diff
@@ -201,7 +204,92 @@ def main():
+            elif len(variants) == 1:
+                variant = variants[0]
+            else:
+                raise ValueError("more than one variant found for selected board, specify -V. "
+                    "Found variants: {}".format(" ".join(sorted(variants))))
+        variant_dir = ''
+        if needs_artifacts:
+            if args.srcbuild:
+                variant_dir = variant
+            else:
+                variant_dir = 'kasli-' + variant
+
+        if not args.partition:
+            args.partition = "gateware bootloader firmware start".split()
+            
+        def artifact_path(this_variant_dir, *path_filename):
+            if args.srcbuild:
+                # source tree - use path elements to locate file
+                return os.path.join(bin_dir, this_variant_dir, *path_filename)
+            else:
+                # flat tree - all files in the same directory, discard path elements
+                *_, filename = path_filename
+                return os.path.join(bin_dir, this_variant_dir, filename)
+        
+        def convert_gateware(bit_filename, header=False):
+            bin_handle, bin_filename = tempfile.mkstemp(
+                prefix="artiq_", suffix="_" + os.path.basename(bit_filename))
+            with open(bit_filename, "rb") as bit_file, \
+                    open(bin_handle, "wb") as bin_file:
+                if header:
+                    bin_file.write(b"\x00"*8)
+                bit2bin(bit_file, bin_file)
+                if header:
+                    magic = 0x5352544d  # "SRTM", see sayma_rtm target
+                    length = bin_file.tell() - 8
+                    bin_file.seek(0)
+                    bin_file.write(magic.to_bytes(4, byteorder="big"))
+                    bin_file.write(length.to_bytes(4, byteorder="big"))
+            atexit.register(lambda: os.unlink(bin_filename))
+            return bin_filename
+            
+        for partition in args.partition:
+            if partition == "gateware":
+                gateware_bin = convert_gateware(
+                    artifact_path(variant_dir, "gateware", "top.bit"))
+                with open(gateware_bin, "rb") as fi:
+                    file = fi.read()
+                    mgmt.flash_write(partition, file)
+            elif partition == "bootloader":
+                bootloader_bin = artifact_path(variant_dir, "software", "bootloader", "bootloader.bin")
+                with open(bootloader_bin, "rb") as fi:
+                    file = fi.read()
+                    mgmt.flash_write(partition, file)
+            elif partition == "firmware":
+                firmware = "runtime"
+                firmware_fbi = artifact_path(variant_dir, "software", firmware, firmware + ".fbi")
+                with open(firmware_fbi, "rb") as fi:
+                    file = fi.read()
+                    mgmt.flash_write(partition, file)
+            elif partition == "start":
```

Comment:
partition?

Reply:
I think gateware/bootloader/firmware is like partitions in flash. I can change the name to `action` just like artiq_flash